js/src/widgets/imageView.js

```diff
@@ -71,20 +78,195 @@
                                  '</div>'
     ].join('')),
 
-    bindEvents: function() {
-      var _this = this;
-      jQuery.subscribe('fitBounds.' + _this.parent.id, function(event, bounds) {
+    listenForActions: function() {
+      var _this = this,
+      firstCanvasId = _this.imagesList[0]['@id'],
+      lastCanvasId = _this.imagesList[_this.imagesList.length-1]['@id'];
+
+      jQuery.subscribe('bottomPanelSet.' + _this.windowId, function(event, visible) {
+        var dodgers = _this.element.find('.mirador-osd-toggle-bottom-panel, .mirador-pan-zoom-controls');
+        var arrows = _this.element.find('.mirador-osd-next, .mirador-osd-previous');
+        if (visible === true) {
+          dodgers.css({transform: 'translateY(-130px)'});
+          arrows.css({transform: 'translateY(-65px)'});
+        } else {
+          dodgers.css({transform: 'translateY(0)'});
+          arrows.css({transform: 'translateY(0)'});
+        }
+      });
+
+      jQuery.subscribe('fitBounds.' + _this.windowId, function(event, bounds) {
         var rect = _this.osd.viewport.imageToViewportRectangle(Number(bounds.x), Number(bounds.y), Number(bounds.width), Number(bounds.height));
         _this.osd.viewport.fitBoundsWithConstraints(rect, false);
       });
 
+      jQuery.subscribe('currentCanvasIDUpdated.' + _this.windowId, function(event, canvasId) {
+        // If it is the first canvas, hide the "go to previous" button, otherwise show it.
+        if (canvasId === firstCanvasId) {
+          _this.element.find('.mirador-osd-previous').hide();
+          _this.element.find('.mirador-osd-next').show();
+        } else if (canvasId === lastCanvasId) {
+          _this.element.find('.mirador-osd-next').hide();
+          _this.element.find('.mirador-osd-previous').show();
+        } else {
+          _this.element.find('.mirador-osd-next').show();
+          _this.element.find('.mirador-osd-previous').show();
+        }
+        // If it is the last canvas, hide the "go to previous" button, otherwise show it.
+      });
+
+      //Related to Annotations HUD
+      jQuery.subscribe('HUD_REMOVE_CLASS.' + _this.windowId, function(event, elementSelector, className) {
+        _this.element.find(elementSelector).removeClass(className);
+      });
+
+      jQuery.subscribe('HUD_ADD_CLASS.' + _this.windowId, function(event, elementSelector, className) {
+        _this.element.find(elementSelector).addClass(className);
+      });
+
+      jQuery.subscribe('HUD_FADE_IN.' + _this.windowId, function(event, elementSelector, duration) {
+        _this.element.find(elementSelector).fadeIn(duration);
+      });
+
+      jQuery.subscribe('HUD_FADE_OUT.' + _this.windowId, function(event, elementSelector, duration, complete) {
+        _this.element.find(elementSelector).fadeOut(duration, complete);
+      });
+      //Related to Annotations HUD
+    },
+
+    bindEvents: function() {
+      var _this = this;
```

Comment:
A lot of the actions on the HUD in ImageView is duplicated in BookView (except for the annotation-specific events).  This is a necessary evil, but the duplication will be removed when we use OSD to manage the switch between views.